R/nested.R

```diff
@@ -23,14 +22,26 @@ compute_parse_data_nested <- function(text) {
 #' Obtain token table from text
 #'
 #' [utils::getParseData()] is used to obtain a flat parse table from `text`.
+#'
+#' Apart from the columns provided by `utils::getParseData()`, the following
+#' columns are added:
+#'
+#'   * A column "short" with the first five characters of "text".
+#'   * A column "pos_id" for (positional id) which can be used for sorting
+#'     (because "id" cannot be used in general). Note that the nth value of this
+#'     column corresponds to n as long as no tokens are inserted.
+#'   * A column "child" that contains the nested subtibbles.
+#'
 #' @param text A character vector.
 #' @return A flat parse table
+#' @importFrom rlang seq2
 tokenize <- function(text) {
   # avoid https://bugs.r-project.org/bugzilla3/show_bug.cgi?id=16041
   parse(text = text, keep.source = TRUE)
   parsed <- parse(text = text, keep.source = TRUE)
   parse_data <- as_tibble(utils::getParseData(parsed, includeText = NA)) %>%
     enhance_mapping_special()
+  parse_data$pos_id <- seq2(1, nrow(parse_data))
```

Comment:
Using `1L` instead of `1` might just keep the data in the integer domain. I see that we later use numeric sometimes (+0.3 and +0.6), but this doesn't need to be the default.

Reply:
Thanks, sure.

Reply:
Ok, thanks. I think `seq2()` creates intervals of integers anyways, 
``` r
class(rlang::seq2(1, 3))
#> [1] "integer"
```
but we can be a bit more explicit if you like.